utilities/src/main/java/io/syndesis/qe/bdd/validation/IrcValidationSteps.java

```diff
@@ -2,131 +2,52 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-import org.apache.commons.lang3.StringUtils;
+import io.syndesis.qe.templates.IrcTemplate;
+import io.syndesis.qe.utils.HttpUtils;
+import io.syndesis.qe.utils.OpenShiftUtils;
+import io.syndesis.qe.utils.TestUtils;
 
-import org.schwering.irc.lib.IRCConnection;
-import org.schwering.irc.lib.IRCEventListener;
-import org.schwering.irc.lib.IRCModeParser;
-import org.schwering.irc.lib.IRCUser;
+import com.google.gson.Gson;
 
-import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 import cucumber.api.java.en.Then;
 import cucumber.api.java.en.When;
-import io.syndesis.qe.TestConfiguration;
-import io.syndesis.qe.utils.TestUtils;
-import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
 public class IrcValidationSteps {
-    private IRCConnection connection;
-    private IrcListener listener;
-
-    @When("^connect IRC listener to channels \"([^\"]*)\"$")
-    public void connectListener(String channels) throws Exception {
-        connection = new IRCConnection(StringUtils.substringBetween(TestConfiguration.openShiftUrl(), "//", ":"), new int[] {31111},
-                null, "checker", "checker", null);
-        listener = new IrcListener();
-        connection.addIRCEventListener(listener);
-        connection.connect();
-        Thread.sleep(15000L);
-        for (String chan : channels.split(",")) {
-            connection.doJoin(chan);
-        }
+    private static final String CONTROLLER_ROUTE = "http://" + OpenShiftUtils.getInstance().routes()
```

Comment:
How are we sure that the route exists before the class is loaded?

Reply:
Only this way, that it is basically the first step of the tests:

https://github.com/syndesisio/syndesis-qe/blob/c442b4ae8c9a32f6317c0a38ef1eb7544c070120/rest-tests/src/test/resources/features/irc.feature#L7-L9